modules/integration/tests-common/admin-clients/src/main/java/org/wso2/bps/integration/common/clients/bpmn/ActivitiRestClient.java

```diff
@@ -0,0 +1,684 @@
+package org.wso2.bps.integration.common.clients.bpmn;
+
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpHost;
+import org.apache.http.HttpResponse;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.methods.*;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.http.util.EntityUtils;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.apache.commons.logging.Log;
+
+import java.io.File;
+
+public class ActivitiRestClient {
+
+	private final static String username = "admin";
+	private final static String password = "admin";
+	private final static String userClaim = "paul";
+	private final static String userDelegate = "will";
+	private final static int port = 9763 ;
+	private final static String preURL = "http://localhost:"+port+"/bpmnrest/";
+	private static final Log log = LogFactory.getLog(ActivitiRestClient.class);
+
+	public void waitForTaskGeneration(){
+		try {
+			Thread.sleep(1000);
+		} catch (InterruptedException ie) {
+
+		}
+
+	}
+
+	//region DeploymentMethods
+	public String[] deployBPMNPackage(String filePath, String fileName) throws Exception{
+		String url = preURL+"repository/deployments";
+
+		HttpHost target = new HttpHost("localhost",port,"http");
+
+		CredentialsProvider credProvider = new BasicCredentialsProvider();
+		DefaultHttpClient httpClient = new DefaultHttpClient();
+		httpClient.getCredentialsProvider().setCredentials(
+				new AuthScope(target.getHostName(),target.getPort()),
+		        new UsernamePasswordCredentials(username,password)
+				);
+			try {
+
+				HttpPost httpPost = new HttpPost(url);
+
+				MultipartEntityBuilder builder = MultipartEntityBuilder.create();
+				builder.addBinaryBody("file", new File(filePath),
+				                      ContentType.MULTIPART_FORM_DATA,fileName);
+				HttpEntity multipart = builder.build();
+				httpPost.setEntity(multipart);
+				HttpResponse response = httpClient.execute(httpPost);
+
+				try {
+					String status = response.getStatusLine().toString();
+					String responseData = EntityUtils.toString(response.getEntity());
+					JSONObject jsonResponseObject = new JSONObject(responseData);
```

Comment:
Defined Status Codes as Constants and used the constants, Fixed in the latest commit.